packages/core/protocols/src/proto/dxos/data.proto

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright 2021 DXOS.org
+//
+
+
+syntax = "proto3";
+
+import "dxos/keys.proto";
+import "dxos/echo/timeframe.proto";
+import "google/protobuf/any.proto";
+
+package dxos.config;
+
+// Requirements:
+// - General purpose mechanism for a client to initiate sync from server.
+// - Use cases: peer start-up from disk; client/service-worker; peer/sync-bot; peer joining from swarm; devtools.
+// - Client initiates subscription which includes the spec for the sync "session" managed by the server.
+// - Server sends stream of events, which update the client's cache.
+// - Client can simultaneously send mutation events.
+// - Only one Space per sync session.
+// - ISSUE: Can the client can have multiple sessions open at once.
+//      - If not how does the server support multiple clients?
+//      - Can different clients request different sync states?
+//
+// Notes:
+// - https://github.com/dxos/dxos/issues/2227
+
+service DataService {
+  rpc CreateSubscription(DataQuery) returns (stream DataEvent);
```

Comment:
Can a query be updated? E.g. more object ids added to the filter?

Reply:
i'm not sure. see my issue about competing subscriptions. the worker's pipeline can only be in one state so if there are multiple subscriptions (e.g. from different clients) then hire can it handle those. similarly if the single subscription is changed hire dues the service know which of your objects to invalidate?